libraries/cmake/source/lief/CMakeLists.txt

```diff
@@ -0,0 +1,388 @@
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV0.cpp"
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV0.tcc"
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV1.cpp"
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV1.tcc"
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV2.cpp"
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV2.tcc"
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV3.cpp"
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV3.tcc"
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV4.cpp"
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV4.tcc"
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV5.cpp"
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV5.tcc"
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV6.cpp"
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV6.tcc"
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV7.cpp"
+    "${library_root}/src/PE/LoadConfigurations/LoadConfigurationV7.tcc"
+    "${library_root}/src/PE/signature/Attribute.cpp"
+    "${library_root}/src/PE/signature/ContentInfo.cpp"
+    "${library_root}/src/PE/signature/Signature.cpp"
+    "${library_root}/src/PE/signature/SignerInfo.cpp"
+    "${library_root}/src/PE/signature/x509.cpp"
+    "${library_root}/src/PE/signature/OIDToString.cpp"
+    "${library_root}/src/PE/signature/SignatureParser.cpp"
+    "${library_root}/src/PE/signature/RsaInfo.cpp"
+    "${library_root}/src/PE/signature/attributes/GenericType.cpp"
+    "${library_root}/src/PE/signature/attributes/ContentType.cpp"
+    "${library_root}/src/PE/signature/attributes/SpcSpOpusInfo.cpp"
+    "${library_root}/src/PE/signature/attributes/MsCounterSign.cpp"
+    "${library_root}/src/PE/signature/attributes/MsSpcNestedSignature.cpp"
+    "${library_root}/src/PE/signature/attributes/MsSpcStatementType.cpp"
+    "${library_root}/src/PE/signature/attributes/PKCS9AtSequenceNumber.cpp"
+    "${library_root}/src/PE/signature/attributes/PKCS9CounterSignature.cpp"
+    "${library_root}/src/PE/signature/attributes/PKCS9MessageDigest.cpp"
+    "${library_root}/src/PE/signature/attributes/PKCS9SigningTime.cpp"
+    )
+    target_include_directories(thirdparty_lief PRIVATE
+    "${library_root}/src/PE/utils/ordinals_lookup_tables"
+    "${library_root}/src/PE/utils/ordinals_lookup_tables_std"
+    "${library_root}/src/PE/signatures/"
+    )
+    if(MSVC)
+      add_definitions(-DNOMINMAX)
+      # Used for the 'and', 'or' ... keywords - See: http://www.cplusplus.com/reference/ciso646/
+      target_compile_options(thirdparty_lief PUBLIC /FIiso646.h)
+      set_property(TARGET thirdparty_lief PROPERTY LINK_FLAGS /NODEFAULTLIB:MSVCRT)
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /bigobj")
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /bigobj")
```

Comment:
Avoid modifying these variables. If compiler flags have to be changed specific to this library, use `target_compile_options` as you did above.
Also `/bigobj` is already present when we compile (https://github.com/osquery/osquery/blob/00d1796d0e06538bb9057f7d9a00a15e2d167ebf/cmake/flags.cmake#L267)

Reply:
fixed, good to know thanks.